framework/validators/CompareValidator.php

```diff
@@ -85,28 +85,28 @@ public function init()
         if ($this->message === null) {
             switch ($this->operator) {
                 case '==':
-                    $this->message = Yii::t('yii', '{attribute} must be repeated exactly.');
+                    $this->message = Yii::t('yii', '{attribute} must be equal to "{compareInternal}".');
```

Comment:
I just read the current code. I don't think any change needs to be made. We already support `{compareAttribute}`, although the default message doesn't use it. So if `{compareValue}` is not appropriate, you can always customize the message to use `{compareAttribute}`.